hack/verify-golint.sh

```diff
@@ -52,7 +52,7 @@ fi
 
 export IFS=$'\n'
 all_packages=(
-	$(go list -e ./... | egrep -v "/(third_party|vendor|staging|generated|clientset_generated)" | sed 's/k8s.io\/kubernetes\///g')
+	$(go list -e ./... | egrep -v "/(third_party|vendor|staging|generated|clientset_generated)" | sed -r "s/^_${KUBE_ROOT//\//\\/}\///g")
```

Comment:
Yeah, if you run outside of GOPATH, go gives up and throws that leading
underscore in to give you a clue.  Pretty useless IMO.  Maybe make the
regex do both, and document it REALLY WELL

`... | sed -e 's|^k8s.io/kubernetes/||' -e "s|^_${KUBE_ROOT}/\?||"`

On Thu, Sep 1, 2016 at 4:22 PM, Davanum Srinivas <notifications@github.com>
wrote:

> In hack/verify-golint.sh
> <https://github.com/kubernetes/kubernetes/pull/31913#discussion_r77271432>
> :
>
> > @@ -52,7 +52,7 @@ fi
> >
> >  export IFS=$'\n'
> >  all_packages=(
> > -	$(go list -e ./... | egrep -v "/(third_party|vendor|staging|generated|clientset_generated)" | sed 's/k8s.io\/kubernetes\///g')
> > +	$(go list -e ./... | egrep -v "/(third_party|vendor|staging|generated|clientset_generated)" | sed -r "s/^_${KUBE_ROOT//\//\\/}\///g")
>
> So if i run from the GO directory thing, yes i see what you see which is
>
> [dims@dims-mac 19:21] ~/go/src/k8s.io/kubernetes ⟩ go list -e ./... | tail -5k8s.io/kubernetes/vendor/gopkg.in/inf.v0k8s.io/kubernetes/vendor/gopkg.in/natefinch/lumberjack.v2k8s.io/kubernetes/vendor/gopkg.in/yaml.v2k8s.io/kubernetes/vendor/k8s.io/heapster/metrics/api/v1/typesk8s.io/kubernetes/vendor/k8s.io/heapster/metrics/apis/metrics/v1alpha1
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31913/files/c3d1c25fa04798b4d6dc2bdf052d1161a88daacd#r77271432>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVOsS6KmOvmQ4o9k-zo_Ya0aVeOqEks5ql15DgaJpZM4JzGw8>
> .
>